llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -12509,6 +12509,34 @@ struct AAIndirectCallInfoCallSite : public AAIndirectCallInfo {
 
 /// ------------------------ Address Space  ------------------------------------
 namespace {
+
+template <typename InstType>
+static bool makeChange(Attributor &A, InstType *MemInst, const Use &U,
+                       Value *OriginalValue, PointerType *NewPtrTy,
+                       bool UseOriginalValue) {
+  if (U.getOperandNo() != InstType::getPointerOperandIndex())
+    return false;
+
+  auto *TTI = A.getInfoCache().getAnalysisResultForFunction<TargetIRAnalysis>(
+      *MemInst->getFunction());
+  if (!TTI)
+    return false;
```

Comment:
Don't need to query this or bail until you see a volatile. Also shouldn't the TTI just be available as context already? And why would it ever fail?

Reply:
It is a pointer so it is safer to check it before use. If it never a `nullptr`, this function interface should return a reference instead of a pointer.